### PR TITLE
Upgrade on code for clarify

### PR DESCRIPTION
**Pull request by LightYagami28**
Hello Alba4k,

I made upgrade on your code.

This is my upgrade

### CODE REVIEW

**Comment by LightYagami28** [comment]
Explanatory Comments: Added comments to clarify the purpose of each section, making it easier to understand.
Emoji in the Title: Included a 🚀 emoji in the project description to give it a modern and visual appeal.
Formatting and Structure: Enhanced indentation and separated various sections with whitespace for better readability and navigation.

**Comment by LightYagami28** [comment]
### Improvements Made

1. **Explanatory Comments**: Added comments to clarify the purpose of each section, making it easier to understand.
   ```nix
   # This Nix expression imports the flake-compat package
   (
     import
     (
       let
         # Load the flake.lock file and parse it as JSON for dependency management
         lock = builtins.fromJSON (builtins.readFile ./flake.lock);
       in
         # Fetch the tarball of flake-compat using the locked revision and its SHA256 hash
         fetchTarball {
           url = "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
           sha256 = lock.nodes.flake-compat.locked.narHash;
         }
     )
     # Specify the current directory as the source for the import
     { src = ./.; }
   )
   .defaultNix
   ```

2. **Consistent Indentation**: Enhanced indentation for better readability.
3. **Descriptive Variable Names**: Maintained clear naming conventions for clarity.

**Comment by LightYagami28** [comment]
### Improvements Made

1. **Enhanced Structure**: Organized sections with clearer headings.
2. **Descriptive Bullets**: Added bullet points for better readability and quick scanning.
3. **Emojis**: Included relevant emojis to enhance visual appeal and emphasize changes.

**Comment by LightYagami28** [comment]
## Changes and Improvements:

- **Comments**: All comments have been translated to English for better understanding.
- **Reorganization**: Some parts of the code have been reorganized for improved readability.
- **Use of `//`**: The union operator `//` is used to merge legacy packages more clearly.
- **System Logic**: The logic for assigning legacy packages based on the current system has been simplified.

**Comment by LightYagami28** [comment]
## Changes and Improvements:

- **Comments**: Added comments to explain the purpose of each section, making it easier to understand the code.
- **Clarity**: The structure remains the same, but the comments help clarify the intent behind each line.
- **Maintainability**: The use of `inherit` statements is preserved for flexibility, allowing for easy updates or changes in the future.

**Comment by LightYagami28** [comment]
### Changes and Improvements Made:

- **Size Constants**: Defined constants for string sizes (`CAPACITY_MAX_LEN`, `STATUS_MAX_LEN`, `BUFFER_SIZE`) to make the code more readable and easier to modify.

- **Memory Management**: Ensured that the strings are properly null-terminated to avoid potential buffer overflow issues.

- **Improved Comments**: Made the comments clearer and more contextual, better explaining the sections of the code.

- **Result Formatting**: Simplified the logic for formatting the result, improving clarity.

- **Use of `tolower`**: Maintained the lowercase conversion for the battery status but slightly restructured the code to enhance readability.

File: flake.nix [hunk 1]
```diff
@@ -1,42 +1,54 @@
...
+      # Supported system architectures
       systems = [
-        "x86_64-linux"
-        "aarch64-linux"
-        "x86_64-darwin"
-        "aarch64-darwin"
+        "x86_64-linux"    # 64-bit Linux
+        "aarch64-linux"   # 64-bit ARM Linux
+        "x86_64-darwin"   # 64-bit macOS
+        "aarch64-darwin"  # ARM-based macOS (M1/M2 chips)
```
Comment (alba4k) [inline flake.nix:46]: Please change this to (Apple M* chips), as it would incluse stuff like M3, the upcoming M4 and any future releases

**Review by alba4k — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 1 above)
Comment (LightYagami28) [reply flake.nix:46]: Yes it's not a problem

**Comment by LightYagami28** [comment]
### Changes and Improvements Made in bios.c :

- **Buffer Size Constant**: Defined a constant (`BUFFER_SIZE`) for easier adjustments and improved code readability.

- **Memory Management**: Removed dynamic memory allocation in favor of stack-allocated buffers for `vendor` and `version`, reducing memory management overhead and potential leaks.

- **String Reading**: Replaced manual reading and null termination with `fgets`, which simplifies the code and ensures safety against buffer overflows.

- **Newline Removal**: Added logic to remove the newline character from the read strings to ensure clean output.

- **Error Handling**: Maintained clarity in error handling with concise checks for data availability.

**Review by alba4k — CHANGES_REQUESTED** [review]
No new modules were added, so the `Modules` title should not have anything below it

This change should be below a `Config Syntax` list

File: Makefile [hunk 2]
```diff
@@ -19,31 +17,42 @@ ifeq ($(OS),Darwin)
...
+# Compile the project
 compile: build
 	meson compile -C build
 
+# Run the compiled binary
 run: compile
-	build/albafetch
+	./build/albafetch
 
+# Debug build
```
Comment (alba4k) [inline Makefile:42]: The debug build is produced by `compile`, this section simply runs it. The comment should be updated to reflect the one above `run`

File: README.md [hunk 3]
```diff
@@ -1,236 +1,208 @@
...
 <details>
+  <summary>Time Comparison</summary>
 
-<summary>Time comparison</summary>
+  Neofetch:
+  ![Neofetch](images/time_neofetch.png)
 
-![neofetch](images/time_neofetch.png)
-![albafetch](images/time_albafetch.png)
+  Albafetch:
```
Comment (alba4k) [inline README.md:51]: albafetch should always be lowercase. Please make sure that this is the case in all instances.

File: meson.build [hunk 4]
```diff
@@ -54,20 +54,21 @@ if host_machine.system() == 'darwin'
   ]
 endif
 
-src_debug = ['src/debug.c']
-src_debug += src
-src += 'src/main.c'
+# Source files for the debug executable
+src_debug = ['src/debug.c'] + src
```
Comment (alba4k) [inline meson.build:59]: No. main.c should not be part of src_debug and should therefore be added at a later time

File: src/info/battery.c [hunk 5]
```diff
@@ -4,72 +4,92 @@
...
+#define BUFFER_SIZE 256
 
-        char *ptr = strstr(buf, "percentage");
-        char *ptr2 = strstr(buf, "status");
-        char *end;
+// Function to get the battery percentage and status (Linux only)
+int battery(char *dest)
+{
+    char capacity[CAPACITY_MAX_LEN + 1] = "";
+    char status[STATUS_MAX_LEN + 1] = "";
```
Comment (alba4k) [inline src/info/battery.c:17]: no +1 is needed since you already set the macros to include the closing \0

**Review by alba4k — CHANGES_REQUESTED** [review]

**Comment by alba4k** [comment]
call me up when this is ready to be merged

**Comment by alba4k** [comment]
Additionally, you may want to define buffer size in a different header (like info.h) instead of per-file as it will be constant

and you would also be able to use it in main.c

File: src/info/battery.c [hunk 6]
```diff
@@ -4,72 +4,92 @@
...
+    char *percentage_ptr = strstr(buf, "percentage");
+    if (percentage_ptr != NULL)
+    {
+        percentage_ptr += 13; // Move past "percentage": "
+        char *end = strchr(percentage_ptr, ',');
+        if (end != NULL)
+        {
+            *end = '\0'; // Null-terminate the string
+            strncpy(capacity, percentage_ptr, CAPACITY_MAX_LEN);
         }
```
Comment (alba4k) [inline src/info/battery.c:35]: Inconsistent standards

I'm using
```c
if(condition) {
    // code
}
```
and you're using 
```c
if (condition)
{
    // code
}
```

Make sure it's consistent everywhere

**Comment by LightYagami28** [comment]
I have replaced old code for new commit

**Comment by LightYagami28** [comment]
I made a fix for a couple of vulnerability in cpu.c:

[{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 63,
	"startColumn": 18,
	"endLineNumber": 63,
	"endColumn": 26
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strchr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 67,
	"startColumn": 22,
	"endLineNumber": 67,
	"endColumn": 30
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 102,
	"startColumn": 22,
	"endLineNumber": 102,
	"endColumn": 30
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 104,
	"startColumn": 22,
	"endLineNumber": 104,
	"endColumn": 30
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 106,
	"startColumn": 22,
	"endLineNumber": 106,
	"endColumn": 30
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 108,
	"startColumn": 22,
	"endLineNumber": 108,
	"endColumn": 30
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 110,
	"startColumn": 22,
	"endLineNumber": 110,
	"endColumn": 30
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 112,
	"startColumn": 22,
	"endLineNumber": 112,
	"endColumn": 30
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/DerefNull",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Dereference of a NULL Pointer: A pointer is possibly assigned the return value of a standard library...",
	"source": "Snyk Code",
	"startLineNumber": 115,
	"startColumn": 10,
	"endLineNumber": 115,
	"endColumn": 13
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 119,
	"startColumn": 26,
	"endLineNumber": 119,
	"endColumn": 34
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 121,
	"startColumn": 31,
	"endLineNumber": 121,
	"endColumn": 39
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strstr after it may already have been freed with...",
	"source": "Snyk Code",
	"startLineNumber": 123,
	"startColumn": 31,
	"endLineNumber": 123,
	"endColumn": 39
},{
	"resource": "/c:/Users/maule/Desktop/Github/albafetch/src/info/cpu.c",
	"owner": "_generated_diagnostic_collection_name_#4",
	"code": {
		"value": "cpp/UseAfterFree",
		"target": {
			"$mid": 1,
			"path": "/scan-using-snyk/snyk-code/snyk-code-security-rules",
			"scheme": "https",
			"authority": "docs.snyk.io"
		}
	},
	"severity": 4,
	"message": "Use After Free: Use after free. cpu_info is used in strncpy after it may already have been freed wit...",
	"source": "Snyk Code",
	"startLineNumber": 127,
	"startColumn": 19,
	"endLineNumber": 127,
	"endColumn": 27
}]

**Pull request closed without merging**